frontend/src/components/ProfileContent.tsx

```diff
@@ -101,13 +173,39 @@ export default function ProfileContent(props: ProfileContentProps) {
                     <div className={classes.introVideo}>Intro Video</div>
                 </Grid>
             </Grid>
-            <Grid container item alignItems="center" className={classes.fieldsContainer} xs={12}>
-                {profileFields.map(({ title, value }: ProfileField) => (
-                    <Grid key={title} item className={classes.field} sm={12} md={6}>
-                        <Typography>{title}: {value}</Typography>
+            { isEditMode ? null :
+                <Grid id={readOnlyContent} container item alignItems="center" className={classes.fieldsContainer} xs={12}>
+                    <Button variant="contained" color="secondary" className={classes.bottomRight} onClick={handleEditBtnClick}>Edit</Button>
+                    <Grid id={readOnlyContent} container item alignItems="center" xs={12}>
+                        {Object.keys(profileFields).map(function (key) {
+                            let field: ProfileField = profileFields[key];
+                            const { title, value } = field;
+                            return (
+                                <Grid key={`${key}-readonlyFieldGrid`} item className={classes.field} sm={12} md={6}>
+                                    <Typography id={`${key}-readonlyFieldText`}>{title}: {value}</Typography>
+                                </Grid>
+                            );
+                        })}
                     </Grid>
-                ))}
-            </Grid>
+                </Grid>
+            }
+            { isEditMode ?
+                <Grid id={editableContent} container item alignItems="center" className={classes.fieldsContainer} xs={12}>
+                    <Grid id={editableContent} container item alignItems="center" xs={12}>
+                        <Grid item className={classes.field} sm={3} md={6}>
+                            <form onSubmit={saveFields}>
+                                <TextField required label={editedProfileFields.name.title} defaultValue={editedProfileFields.name.value} onChange={(e) => { setName(e.target.value) }} />
+                                <TextField disabled label={editedProfileFields.username.title} defaultValue={editedProfileFields.username.value} onChange={(e) => { setUsername(e.target.value) }} />
+                                <TextField required label={editedProfileFields.age.title} defaultValue={editedProfileFields.age.value} onChange={(e) => { setAge(e.target.value) }} />
+                                <TextField label={editedProfileFields.gender.title} defaultValue={editedProfileFields.gender.value} onChange={(e) => { setGender(e.target.value) }} />
```

Comment:
Since the `onChange` is just passing the argument through, I believe you can just use:
`onChange={setThing}`

Reply:
I thought so too, and maybe there is a clever way to do this, but it throws:
TS2322: Type 'Dispatch<SetStateAction<string>>' is not assignable to type '(event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => void'.

Reply:
ahhh

Reply:
In this instance, I think using [`e.currentTarget.value` is more accurate than `e.target.value`](https://developer.mozilla.org/en-US/docs/Web/API/Event/Comparison_of_Event_Targets). In general, this is usually the case because currentTarget will always be the element this listener is attached to.